LykkeApi2/Controllers/FeaturesController.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Core.Repositories;
+using LykkeApi2.Models.Features;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace LykkeApi2.Controllers
+{
+    [Route("api/features")]
+    [ApiController]
+    public class FeaturesController : Controller
+    {
+        private readonly IFeaturesRepository _featuresRepository;
+
+        public FeaturesController(IFeaturesRepository featuresRepository)
+        {
+            _featuresRepository = featuresRepository;
+        }
+
+        /// <summary>
+        /// Get all features and their global enabled/disabled status.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet]
+        [ProducesResponseType(typeof(IDictionary<string, bool>), (int) HttpStatusCode.OK)]
+        public async Task<IActionResult> Get()
+        {
+            return Ok(await _featuresRepository.GetAll(null));
+        }
+        
+        /// <summary>
+        /// Get all features and their enabled/disabled status for specific client.
+        /// If there is no entry for specific client, global setting will be used.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(typeof(IDictionary<string, bool>), (int) HttpStatusCode.OK)]
+        public async Task<IActionResult> GetForClient(string clientId)
+        {
+            return Ok(await _featuresRepository.GetAll(clientId));
+        }
+        
+        /// <summary>
+        /// Sets enabled/disabled status for a feature for a specific client
+        /// If client id is not specified, then the feature will be toggled globally
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost]
+        [SwaggerOperation("SetFeature")]
+        public async Task<IActionResult> SetFeature([FromBody] ToggleFeatureRequest request)
```

Comment:
it's not needed for this kind of API. It will be implemented in Back Office separately

Reply:
`+`